apps/web/components/StepperFSM/StepperFSMUtils.tsx

```diff
@@ -0,0 +1,386 @@
+    step.stepType.toLowerCase().includes('radio')
+  ) {
+    return STEP_TYPES.QUESTION_RADIO
+  }
+  if (
+    step.stepType &&
+    step.stepType.toLowerCase().includes('question') &&
+    step.stepType.toLowerCase().includes('dropdown')
+  ) {
+    return STEP_TYPES.QUESTION_DROPDOWN
+  }
+  if (step.stepType && step.stepType.toLowerCase().includes('answer')) {
+    return STEP_TYPES.ANSWER
+  }
+  return null
+}
+
+const getStepOptions = (
+  step: Step,
+  lang = 'en',
+  optionsFromNamespace: { slug: string; data: [] }[] = null,
+): StepOption[] => {
+  if (!step || step.config === '') return []
+  const stepConfig: StepConfig = JSON.parse(step.config) as StepConfig
+
+  if (stepConfig.optionsFromSource && optionsFromNamespace) {
+    const stepOptions = optionsFromNamespace.find(
+      (value) => value.slug === step.slug,
+    )
+    if (!stepOptions || !stepOptions.data) return []
+    return stepOptions.data.map((o) => {
+      const {
+        labelFieldEN,
+        labelFieldIS,
+        optionSlugField,
+        transitions,
+      } = stepConfig.optionsFromSource
+      const label = lang === 'en' ? o[labelFieldEN] : o[labelFieldIS]
+      let stepTransition = ''
+
+      transitions.sort((a, b) => {
+        if (!a.priority && !b.priority) return 0
+        if (!a.priority) return 1
+        if (!b.priority) return -1
+        return a.priority - b.priority
+      })
+
+      for (const { criteria, transition, criteriaExclude } of transitions) {
+        const everyCriteriaMatches = Object.keys(criteria).every(
+          (key) => o[key] === criteria[key],
+        )
+        const everyExclusionCriteriaMatches =
+          !criteriaExclude ||
+          Object.keys(criteriaExclude).every(
+            (key) => o[key] !== criteriaExclude[key],
+          )
+
+        if (everyCriteriaMatches && everyExclusionCriteriaMatches) {
+          stepTransition = transition
+          break
+        }
+      }
+      return {
+        label: label,
+        transition: stepTransition,
+        slug: o[optionSlugField],
+      }
+    })
+  }
+
+  if (!stepConfig.options) return []
+
+  return stepConfig.options.map((o) => {
+    const label = lang === 'en' ? o.labelEN : o.labelIS
+    return {
+      label: label,
+      transition: o.transition,
+      slug: o.optionSlug,
+    }
+  })
+}
+
+const getStepOptionsSourceNamespace = (step: Step): string => {
+  if (!step || step.config === '') return ''
+  const stepConfig: StepConfig = JSON.parse(step.config) as StepConfig
+  if (stepConfig.optionsFromSource)
+    return stepConfig.optionsFromSource.sourceNamespace
+  return ''
+}
+
+const stepContainsQuestion = (step: Step) => {
+  return (
+    step.subtitle &&
+    step.subtitle.length > 0 &&
+    step.subtitle[0].__typename === 'Html' &&
+    step.subtitle[0].document.content.length > 0 &&
+    step.subtitle[0].document.content[0].content &&
+    step.subtitle[0].document.content[0].content.length > 0
+  )
```

Comment:
Assuming step is defined, isn't it enough to do:
`step?.subtitle[0]?.document?.content[0]?.content?.length > 0`?
![image](https://user-images.githubusercontent.com/77672665/155724955-c371e288-59a9-43b8-af70-cacb1df1e02d.png)

Reply:
Along with the `html` typename check, I assume